src/agent/src/image_rpc.rs

```diff
@@ -119,29 +363,48 @@ impl ImageService {
     }
 
     async fn pull_image(&self, req: &image::PullImageRequest) -> Result<String> {
+        env::set_var("OCICRYPT_KEYPROVIDER_CONFIG", KEYPROVIDER_PATH);
+
         let image = req.get_image();
-        let mut cid = req.get_container_id();
+        let mut cid = req.get_container_id().to_string();
 
         if cid.is_empty() {
             let v: Vec<&str> = image.rsplit('/').collect();
             if !v[0].is_empty() {
-                cid = v[0]
+                cid = v[0].replace(":", "_");
```

Comment:
I don't understand why this is needed.

Reply:
when cid is empty, since we use the image info as part of container image store path like:
/tmp/busybox:latest/image_oci/
when umoci upack the container image, it will also parase colon and be missleading the tag info

Reply:
But doesn't that imply a bug in umoci? Either way, please add a comment in the code as it's not obvious why this rename is required.

Reply:
Yes, it's a umoci bug which shoud parse the tag from the righ, added the comment now.